pkg/logs/logs.go

```diff
@@ -68,12 +70,12 @@ func start(getAC func() *autodiscovery.AutoConfig, serverless bool, logsChan cha
 
 	// setup the server config
 	httpConnectivity := config.HTTPConnectivityFailure
-	if endpoints, err := config.BuildHTTPEndpoints(); err == nil {
+	if endpoints, err := config.BuildHTTPEndpoints(intakeTrackType, intakeProtocol); err == nil {
 		httpConnectivity = http.CheckConnectivity(endpoints.Main)
 	}
-	endpoints, err := config.BuildEndpoints(httpConnectivity)
+	endpoints, err := config.BuildEndpoints(httpConnectivity, intakeTrackType, intakeProtocol)
 	if serverless {
-		endpoints, err = config.BuildServerlessEndpoints()
+		endpoints, err = config.BuildServerlessEndpoints(intakeTrackType, intakeProtocol)
```

Comment:
for some historical reason (which we hope to change in the future), the Intake backend considers that the json send by the agent is different than by lambda. So
- the normal HTTP endpoint: we set the intakeProtocol to `agent-json`
- the serverless Endpoint: we don't set the intakeProtocol.